### PR TITLE
[Experimental] Avoid unneeded initialization in buffer_vector::resize.

### PR DESCRIPTION
**Pull request by vng**
Needs good manual testing in case if someone relied on implicit buffer zero initialization.

### CODE REVIEW

**Review by biodranik — APPROVED** [review]

File: base/buffer_vector.hpp [hunk 1]
```diff
@@ -34,6 +34,17 @@ template <class T, size_t N> class buffer_vector
     std::move(rhs.m_static, rhs.m_static + rhs.m_size, m_static);
   }
 
+  void SetStaticSize(size_t newSize)
+  {
+    if constexpr (std::is_destructible<T>::value)
+    {
+      // Call destructors for old elements.
+      for (size_t i = newSize; i < m_size; ++i)
+        m_static[i] = T();
```
[the remark this answers is not in the document]
Comment (vng) [reply base/buffer_vector.hpp:43]: Это плохо. В буфере должны находится валидные объекты. Может когда нибудь мы перейдем на какой boost.

File: base/buffer_vector.hpp [hunk 2]
```diff
@@ -34,6 +34,17 @@ template <class T, size_t N> class buffer_vector
     std::move(rhs.m_static, rhs.m_static + rhs.m_size, m_static);
   }
 
+  void SetStaticSize(size_t newSize)
+  {
+    if constexpr (std::is_destructible<T>::value)
+    {
+      // Call destructors for old elements.
```
[the remark this answers is not in the document]
Comment (vng) [reply base/buffer_vector.hpp:41]: хз, у нас в основном он используется для _простых_ типов

File: drape/bidi.cpp [hunk 3]
```diff
@@ -63,35 +63,30 @@ StringT log2vis(StringT const & str)
   if (direction == UBIDI_LTR || direction == UBIDI_NEUTRAL)
     return ResultString();
 
-  uint32_t const buffSize = usize * 2;
-  buffer_vector<UChar, 256> buff(buffSize, 0);
+  int32_t const buffSize = usize * 2;
```
[the remark this answers is not in the document]
Comment (vng) [reply drape/bidi.cpp:66]: ICU работает с  int32_t

File: geometry/point2d.hpp [hunk 4]
```diff
@@ -22,8 +20,7 @@ class Point
 
   T x, y;
 
-  constexpr Point() : x(T()), y(T()) {}
-
+  Point() = default;
```
[the remark this answers is not in the document]
Comment (vng) [reply geometry/point2d.hpp:23]: Предидущий вызывал явную zero-initialization, т.е. {0, 0} Это лишнее и вредное для клиентского кода.
Хз зачем тут constexpr.

[the remark this answers is not in the document]
Comment (biodranik) [reply geometry/point2d.hpp:23]: Полезная штука, можно лепить математику с точками в компайл тайме.
